platform/security/core/security-core-impl/src/test/groovy/ddf/security/samlp/impl/EntityInformationSpec.groovy

```diff
@@ -0,0 +1,292 @@
+        authnRequest = Mock(AuthnRequest) {
+            getProtocolBinding() >> Binding.HTTP_POST.uri
+        }
+        serviceInfo = entityInfo.getAssertionConsumerService(authnRequest, Binding.HTTP_REDIRECT)
+
+        then:
+        serviceInfo.binding != null
+        Binding.HTTP_POST.equals(serviceInfo.binding)
+
+        when: "authnreq has binding and not supported, use specified preferred binding"
+        authnRequest = Mock(AuthnRequest) {
+            getProtocolBinding() >> Binding.HTTP_ARTIFACT.uri
+        }
+        serviceInfo = entityInfo.getAssertionConsumerService(authnRequest, Binding.HTTP_POST)
+
+        then:
+        serviceInfo.binding != null
+        Binding.HTTP_POST.equals(serviceInfo.binding)
+    }
+
+    def "check assertionconsumerservice with half supported metadata"() {
+        setup:
+        MetadataConfigurationParser mcp = new MetadataConfigurationParser(
+                [fakeSpHalfSupportedMetadata])
+        EntityDescriptor ed = mcp.entryDescriptions.find {true}.value
+        EntityInformation entityInfo = new EntityInformation.Builder(ed, DEFAULT_BINDINGS).build()
+
+        when:
+        def serviceInfo = entityInfo.getAssertionConsumerService(null, null)
+
+        then: "should be only supported binding (post)"
+        serviceInfo.binding != null
+        Binding.HTTP_POST.equals(serviceInfo.binding)
+    }
+
+    def "check assertionconsumerservice with a default"() {
+        setup:
+        MetadataConfigurationParser mcp = new MetadataConfigurationParser(
+                [fakeSpHalfSupportedMetadata])
+        EntityDescriptor ed = mcp.entryDescriptions.find {true}.value
+        EntityInformation entityInfo = new EntityInformation.Builder(ed, DEFAULT_BINDINGS).build()
+
+        thisIsNotTheDirtyCodeYouAreLookingFor(entityInfo)
+
+        when:
+        def serviceInfo = entityInfo.getAssertionConsumerService(null, null)
+
+        then:
+        serviceInfo.binding != null
+        Binding.HTTP_REDIRECT.equals(serviceInfo.binding)
+        serviceInfo.url.contains("default.com")
+    }
+
+    def "parse sample sp metadata"() {
+        setup:
+        MetadataConfigurationParser mcp = new MetadataConfigurationParser([sampleSPMetadata])
+        EntityDescriptor ed = mcp.entryDescriptions.find {true}.value
+
+        when:
+        EntityInformation entityInformation = new EntityInformation.Builder(ed, DEFAULT_BINDINGS)
+                .build()
+
+        then:
+        notThrown(Exception)
+    }
+
+    EntityInformation getFakeEntityInfo() {
+        MetadataConfigurationParser mcp = new MetadataConfigurationParser([fakeSpMetadata])
+        EntityDescriptor ed = mcp.entryDescriptions.find {true}.value
+        return new EntityInformation.Builder(ed, DEFAULT_BINDINGS).build()
+    }
+
+    private static void thisIsNotTheDirtyCodeYouAreLookingFor(EntityInformation entityInfo) {
+        // Did I say you could look here
```

Comment:
After discussion, this makes sense to do since the alternatives introduce code smell and this is only for testing purposes. Just add informative comments instead explaining this section